src/lib/builtins.pl

```diff
@@ -123,24 +205,38 @@
 
 % control operators.
 
+%% fail.
+%
+% A predicate that always fails
 fail :- '$fail'.
 
 
 :- meta_predicate \+(0).
 
+%% \+(Goal)
+%
+% Succeeds if Goal fails
 \+ G :- call(G), !, false.
 \+ _.
 
-
+%% \=(?X, ?Y)
+%
+% Succeeds if X and Y can't be unified
 X \= X :- !, false.
 _ \= _.
 
 
 :- meta_predicate once(0).
 
+%% once(Goal)
+%
+% Execute Goal (like call/1) but exactly once, ignoring any kind of alternative solutions the original predicate
+% could have generated.
 once(G) :- call(G), !.
 
-
+%% repeat.
+%
+% This predicate enters an infinite loop, always succeeding and generating infinite choice points
```

Comment:
It does not generate "infinite" choice points, we have neither time nor space available in this universe to generate an infinite amount of anything. One could maybe say it succeeds *arbitrarily often*.